### ISSUE TITLE
Unused locals in catch statements do not cause compiler warnings with --noUnusedLocals

### ISSUE THREAD

**Issue by calebsander** [Microsoft/TypeScript#27518]
**TypeScript Version:** 3.2.0-dev.20181002

**Search Terms:** catch unused error local variable

**Code**

`test.ts`:
```ts
{
	function somethingThatThrows() {
		throw new Error
	}
	try { somethingThatThrows() }
	catch (e) {} // no compiler error even though `e` is never read

	// In contrast:
	const e = new Error // errors because `e` is never read
}
```

Command: `tsc --noUnusedLocals test.ts`

**Expected behavior:** Statements of the form `catch (e) { /*...*/ }` should error under `--noUnusedLocals` if `e` is never read. If `e` is not needed, `catch { /*...*/ }` can be used instead.

**Actual behavior:**

**Playground Link:** N/A

**Comment by RyanCavanaugh** [comment]
Elidable catch arguments were added after `noUnusedLocals` and we didn't want to cause a breaking change.

**Comment by qwelias** [comment]
@RyanCavanaugh was adding `noUnusedElidables` considered here?

**Comment by JasonGore** [comment]
It's 5 years after this has been marked as intended behavior and we find ourselves implementing custom eslint rules to enforce this, which is something that the compiler should arguably already be doing for consistency reasons with other unused local patterns. It feels like there should be some way to guarantee consistency of the compiler here long-term aside from any concerns of breaking changes rolling such consistency patterns out.
